### ISSUE TITLE
Doesn't seem to be published to npm correctly

### ISSUE THREAD

**Issue by DanielRosenwasser** [itsFrank/vue-typescript#1]
Hey there!

I noticed a few things that looked like issues in terms of publishing TypeScript to npm.
1. You're publishing `.ts` files instead of `.js` and `.d.ts`, so your users are being forced to compile in `node_modules` (which isn't great for `.js` users).
2. Your 'main' is `index.ts`, but TypeScript only automatically looks at `typings` (and in 2.0, `types`).
3. You're publishing your `typings` folder (and have it committed in git too). This can cause issues if other users have their own versions of `node.d.ts`

Here's a page with a little bit of detail, though it should be getting fleshed out more: https://www.typescriptlang.org/docs/handbook/typings-for-npm-packages.html

Let me know if you need any help!

**Comment by itsfrank** [comment]
Thanks, this is the first time i'm publishing a module. I did do some reading, but i figured it would be better to leave it in typescript so that if users want to modify some behavior or read the source they could so so with non-generated code (maybe im better pointing package.json to js and leaving the ts in src?). Also, i published typings simply to include the vue typings, but i guess i should leave that up to the user.

I was planning on working on it today, so ill make fixing these issues my first priority, if you wouldn't mind, once i get them fixed in a few hours, could you check it over a second time and let me know if it makes more sense? Thanks!

**Comment by itsFrank** [comment]
I was planning on working on it today, if you wouldn't mind, once i get to fixing these issues in a few hours, could you check it over a second time and let me know if it makes more sense? Thanks!

**Comment by itsfrank** [comment]
So I've followed your advice and changed a few things:
- First i compiled everything to js in the lib directory, pointed my package.json main to "lib/index.js" and added the typings field to pint to "lib/index.d.ts". In lib i include compiled js, definition files as well as sourcemaps (dont know if this is useful or not).
- Then, I removed postinstall "typings install", users will have to provide their own vue typings (although it works fin without, just sets the options types to any instead of the vuejs options types). typings and typings.json are both in .npmignore (however i left them in git, because my circleCI buidls fail without them)

the module folder will now look like this:

vue-typescript
&nbsp;&nbsp;&nbsp;&nbsp; /lib - precompiled js, definitons and sourcemaps
&nbsp;&nbsp;&nbsp;&nbsp; /src - source typescript files, so user can inspect the code
&nbsp;&nbsp;&nbsp;&nbsp; .npmignore
&nbsp;&nbsp;&nbsp;&nbsp; LICENSE
&nbsp;&nbsp;&nbsp;&nbsp; package.json
&nbsp;&nbsp;&nbsp;&nbsp; README.md

I'm debating leaving typings.json and my tsconfig file so that if users want to make quick fixes, they can run `typings install` and `npm run build` to apply the fixes they made to the source files. 

If i forgot anything or you still see issues with the package let me know. I'll probably be working on adding @Watch tonight and writing better tests for @Props

Thanks again for pointing this out
